docker/fetch_tezos_sources.py

```diff
@@ -16,15 +16,12 @@
     [
         "git",
         "clone",
-        "--branch",
-        octez_version,
         "https://gitlab.com/tezos/tezos.git",
-        "--depth",
-        "1",
```

Comment:
There's a reason this was done like this, it takes much less space and I suppose less time to clone only one branch, instead of the whole repo.

Also, the commit doesn't really explain what is causing the issue, was this git command? Why? how does using `git checkout` solve the issue? :thinking: